src/dist/replication/meta_server/meta_bulk_load_service.cpp

```diff
@@ -1144,7 +1157,24 @@ void bulk_load_service::on_control_bulk_load(control_bulk_load_rpc rpc)
                                    false));
     } break;
     case bulk_load_control_type::BLC_RESTART:
-    // TODO(heyuchen): TBD
+        if (app_status != bulk_load_status::BLS_PAUSED) {
+            derror_f(
+                "restart bulk load for app({}) failed, can not restart bulk load with status({})",
+                app_name,
+                dsn::enum_to_string(app_status));
+            response.err = ERR_INVALID_STATE;
+            response.__set_hint_msg(
+                fmt::format("app({}) status={}", app_name, dsn::enum_to_string(app_status)));
```

Comment:
It's never a good practice to use server-side logging for user-facing errors. The user, usually the system admin, won't dig into the logs until he falls into trouble to understand the error. Why not just set the full description in the `hint_msg`?

```suggestion
            response.err = ERR_INVALID_STATE;
            response.__set_hint_msg(
                fmt::format("restart bulk load for app({}) failed, can not restart bulk load with status({})",
                app_name, dsn::enum_to_string(app_status));
```